expressions.go

```diff
@@ -1003,6 +1025,68 @@ func checkBoolExpType(op BooleanOperation, lhs Expression, rhs interface{}, inve
 	return boolean{op: op, lhs: lhs, rhs: rhs}
 }
 
+
+type (
+	RangeOperation  int
+	RangeExpression interface {
+		Expression
+		//Returns the operator for the expression
+		Op() RangeOperation
+		//The left hand side of the expression (e.g. I("a")
+		Lhs() Expression
+		//The right hand side of the expression could be a primitive value, dataset, or expression
+		Rhs1() interface{}
```

Comment:
So Im torn on this you could go a couple of different ways 

1. I think Rhs1 and 2 could be joined into a single slice of what would always be two values here and you could keep with the pattern on Lhs and Rhs and change the between methods to take variadic args but that could get messy.

2. Create a new Range type `I("a").Between(Range(1,2))` which could still allow you to get the single Rhs and allow you to include between and notbetween in the `goqu.Ex` code which was commented out above.

What do you think? I lean towards option 2.